File: Codigo de los 4 lenguajes/C#/EjemploMetodosSinParametros.cs
using System;

public class Maquina
{
    private bool Encendida = false;
    public void Encender(){ Encendida = true; Console.WriteLine("Máquina encendida"); }
    public void Apagar(){ Encendida = false; Console.WriteLine("Máquina apagada"); }
    public bool EstaEncendida(){ return Encendida; }
    public void Reiniciar(){ Console.WriteLine("Reiniciando..."); Encendida=false; Encendida=true; }
    public void Detener(){ Console.WriteLine("Máquina detenida"); }
}

public class EjemploMetodosSinParametros
{
    public void Mostrar()
    {
        Console.WriteLine(" Ejemplo: Métodos sin parámetros ");
        Maquina m = new Maquina();
        m.Encender();
        m.Detener();
        Console.WriteLine("Estado encendida?: " + m.EstaEncendida());
        m.Reiniciar();
        m.Apagar();
        Console.WriteLine();
    }
}

File: Codigo de los 4 lenguajes/js/EjemploMetodosSinParametros.js
// EjemploMetodosSinParametros.js
export default class EjemploMetodosSinParametros {
  mostrar() {
    console.log("=== Ejemplo: Métodos sin parámetros ===");

    class Maquina {
      constructor() { this.encendida = false; }
      encender() { this.encendida = true; console.log("Máquina encendida"); }
      apagar() { this.encendida = false; console.log("Máquina apagada"); }
      estaEncendida() { return this.encendida; }
      reiniciar() { console.log("Reiniciando..."); this.encendida = false; this.encendida = true; }
      detener() { console.log("Máquina detenida"); }
    }

    const m = new Maquina();
    m.encender();
    m.detener();
    console.log("Estado encendida?:", m.estaEncendida());
    m.reiniciar();
    m.apagar();
    console.log();
  }
}
